drivers/include/periph/rtc.h

```diff
@@ -50,8 +50,9 @@ void rtc_init(void);
  *
  * @param[in] time          Pointer to the struct holding the time to set.
  *
- * @return  0 for success
- * @return -1 an error occurred
+ * @retval  0               Success
+ * @retval -EINVAL          Either @p time is `NULL`, or does not specify a valid time
+ * @retval -1               An error occurred while setting the time
```

Comment:
Here and below:

What about `-EIO` instead of unspecified -1 which? -1 stands for `-EPERM` (operation not permitted). Setting and retrieving the time has probably something to do with I/O operation to the RTC module.

Alternatives could be

| Error Code | Output of `strerror(errno)`                    |
|:---------- |:---------------------------------------------- |
| `EPERM`    | Operation not permitted                                  |
| `EIO`   |  I/O error                              |
| `EACCES`     | Permission denied |